riscv/arith.h

```diff
@@ -32,7 +32,7 @@ inline int64_t mulh(int64_t a, int64_t b)
 {
   int negate = (a < 0) != (b < 0);
   uint64_t res = mulhu(a < 0 ? -a : a, b < 0 ? -b : b);
-  return negate ? ~res + (a * b == 0) : res;
+  return negate ? ~res + ((uint64_t)a * (uint64_t)b == 0) : res;
```

Comment:
please explain why to change this?

Spike should be golden. If Spike has bugs, we should raise issues in the upstream instead of fixing them in OpenXiangShan only.